.github/workflows/java-versions.yml

```diff
@@ -38,4 +39,5 @@ jobs:
       - name: Test cross Java versions compatibility
         run: |
           ./gradlew --version
-          ./gradlew --stacktrace build compatTestJava${{ matrix.jdk }}
+          ./gradlew -q javaToolchains
+          ./gradlew --stacktrace -PnexusPublishPlugin.test.java=${{ matrix.testJdk }} build compatTestJava${{ matrix.jdk }}
```

Comment:
To avoid "javaauto" in stutter, we could add testJdk matrix elements manually, adjusting "auto" version to a given JDK version.
Also a shell command could replace "auto" with a proper version.

However, I don't know which of this 3 options is best :-/